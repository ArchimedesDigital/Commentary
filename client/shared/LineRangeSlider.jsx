import { debounce } from 'throttle-debounce';
import '../../node_modules/ion-rangeslider/js/ion.rangeSlider.js';
import '../../node_modules/ion-rangeslider/css/ion.rangeSlider.css';
import '../../node_modules/ion-rangeslider/css/ion.rangeSlider.skinFlat.css';

LineRangeSlider = React.createClass({

	propTypes: {
		handleChangeLineN: React.PropTypes.func,
		lineFrom: React.PropTypes.number,
		lineTo: React.PropTypes.number,
	},

	componentDidMount() {
		$('#line-range').ionRangeSlider({
			type: 'double',
			min: 1,
			max: 909,
			grid: true,
			prettify_enabled: true,
			prettify_separator: ',',
			prefix: 'Line: ',
			values_separator: ' to ',
			onChange: debounce(500, this.props.handleChangeLineN),
			from: this.props.lineFrom,
			to: this.props.lineTo,
		});
	},

	render() {
		return (
			<div id="line-range" />
		);
	},

});
