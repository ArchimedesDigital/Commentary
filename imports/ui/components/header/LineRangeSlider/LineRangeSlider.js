import { debounce } from 'throttle-debounce';
import 'ion-rangeslider/js/ion.rangeSlider.js';
import 'ion-rangeslider/css/ion.rangeSlider.css';
import 'ion-rangeslider/css/ion.rangeSlider.skinFlat.css';

const id = 'line-range';

class LineRangeSlider extends React.Component {

	componentDidMount() {

		const { handleChangeLineN, lineFrom, lineTo } = this.props;
		$(`#${id}`).ionRangeSlider({
			type: 'double',
			min: 1,
			max: 909,
			grid: true,
			prettify_enabled: true,
			prettify_separator: ',',
			prefix: 'Line: ',
			values_separator: ' to ',
			onChange: debounce(500, handleChangeLineN),
			from: lineFrom,
			to: lineTo,
		});
	}

	render() {
		return <input id={id} />;
	}

}
LineRangeSlider.propTypes = {
	handleChangeLineN: React.PropTypes.func.isRequired,
	lineFrom: React.PropTypes.number,
	lineTo: React.PropTypes.number,
};
LineRangeSlider.defaultProps = {
	lineFrom: null,
	lineTo: null,
};

export default LineRangeSlider;
