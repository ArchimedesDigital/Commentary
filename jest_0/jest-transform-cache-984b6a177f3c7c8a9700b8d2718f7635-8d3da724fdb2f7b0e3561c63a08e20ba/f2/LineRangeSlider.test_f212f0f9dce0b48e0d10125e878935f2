/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_reactTestRenderer = require('react-test-renderer'); /* istanbul ignore next */var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);


var /* istanbul ignore next */_LineRangeSlider = require('./LineRangeSlider'); /* istanbul ignore next */var _LineRangeSlider2 = _interopRequireDefault(_LineRangeSlider);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('LineRangeSlider', function () {
	it('renders correctly', function () {

		var tree = /* istanbul ignore next */_reactTestRenderer2.default.
		create( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_LineRangeSlider2.default, /* istanbul ignore next */{ handleChangeLineN: function /* istanbul ignore next */handleChangeLineN() {} })).
		toJSON();
		expect(tree).toMatchSnapshot();
	});
}); // component: