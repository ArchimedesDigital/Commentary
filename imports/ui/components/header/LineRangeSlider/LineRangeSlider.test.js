import React from 'react';
import renderer from 'react-test-renderer';

// component:
import LineRangeSlider from './LineRangeSlider';

describe('LineRangeSlider', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<LineRangeSlider handleChangeLineN={() => {}} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
