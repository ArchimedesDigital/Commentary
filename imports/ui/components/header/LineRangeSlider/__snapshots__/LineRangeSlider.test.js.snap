// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineRangeSlider renders correctly 1`] = `
<input
  id="line-range"
/>
`;
